### ISSUE TITLE
Xenial not able to restore first backed up file with encryption_password

### ISSUE THREAD

**Issue by grke** [grke/burp#640]
From the mailing list:

Hi Graham

On 2017-10-31 11:49, Graham Keeling wrote:
> I have uploaded burp-2.1.22 to sourceforge.
> Here is the link: http://sourceforge.net/projects/burp/files/

Thanks!

Unfortunately, as with 2.1.20, our tests still fail when trying to restore
backups that were created with an encryption_password.

How to reproduce:

* Create new, clean Xenial machine as burp-server with this
  configuration (/etc/burp/burp-server.conf)

```
# THIS CONFIGURATION IS MANAGED BY PUPPET

ca_burp_ca = /usr/sbin/burp_ca
ca_conf = /etc/burp/CA.cnf
ca_name = burpCA
ca_server_name = burpserver.vagrant.dev
client_can_delete = 0
client_can_force_backup = 0
client_can_list = 1
client_can_restore = 1
client_can_verify = 1
clientconfdir = /etc/burp/clients
directory = /var/lib/burp
group = burp
hardlinked_archive = 0
keep = 7
keep = 4
max_children = 10
max_status_children = 5
mode = server
pidfile = /tmp/burp.server.pid
port = 4971
ssl_cert = /var/lib/burp/ssl_cert-server.pem
ssl_cert_ca = /var/lib/burp/ssl_cert_ca.pem
ssl_dhfile = /var/lib/burp/dhfile.pem
ssl_key = /var/lib/burp/ssl_cert-server.key
status_port = 4972
stdout = 0
syslog = 1
timer_arg = 11h
timer_arg = Mon,Tue,Wed,Thu,Fri,Sat,Sun,22,23,00,01,02,03,04,05,06,07
timer_script = /usr/local/bin/burp_timer_script
umask = 0022
user = burp
version_warn = 1
working_dir_recovery_method = resume
dedup_group = global
restore_client = burpserver.vagrant.dev
ssl_compression = zlib0
status_address = ::
```
```
* Create new, clean Xenial machine as burp-client with this
  configuration (/etc/burp/burp.conf)

# THIS CONFIGURATION IS MANAGED BY PUPPET

ca_burp_ca = /usr/sbin/burp_ca
ca_csr_dir = /var/lib/burp-burp/ssl
cname = burpclient.vagrant.dev
cross_all_filesystems = 0
cross_filesystem = /home
mode = client
password = <random-10-chars>
pidfile = /tmp/burp.client.burp.pid
port = 4971
progress_counter = 1
server = burpserver.vagrant.dev
server_can_restore = 0
ssl_cert = /var/lib/burp-burp/ssl/ssl_cert-client.pem
ssl_cert_ca = /var/lib/burp-burp/ssl/ssl_cert_ca.pem
ssl_key = /var/lib/burp-burp/ssl/ssl_cert-client.key
ssl_peer_cn = burpserver.vagrant.dev
stdout = 1
syslog = 0
dedup_group = global
encryption_password = <random-30-chars>
include = /boot/grub
include = /etc
include = /home
include = /root
include = /usr/local
include = /var/backups
include = /var/lib/dpkg
include = /var/log
include = /var/spool
exclude = /etc/network/interfaces.dynamic.d
exclude = /proc
exclude = /var/spool/postfix/defer
exclude = /var/spool/postfix/deferred
nobackup = .nobackup
protocol = 1
status_port = 4972
backup_script_reserved_args = 0
backup_script = /bin/run-parts
backup_script_post_arg = --report
backup_script_post_arg = --regex
backup_script_post_arg = '.*'
backup_script_post_arg = /usr/share/burp/post-backup
backup_script_pre_arg = --report
backup_script_pre_arg = --regex
backup_script_pre_arg = '.*'
backup_script_pre_arg = /usr/share/burp/pre-backup
```
```
* Create backup:
** Client: sudo burp -a t
** Server: sudo touch burp /var/lib/burp/burpclient.vagrant.dev/backup
** Client: sudo burp -a t
** Client: sudo burp -a L -r '^/etc/hosts$'
2017-11-03 17:29:30 +0100: burp[25763] Connecting to
burpserver.vagrant.dev:4971
2017-11-03 17:29:30 +0100: burp[25763] auth ok
2017-11-03 17:29:30 +0100: burp[25763] Server version: 2.1.22
2017-11-03 17:29:30 +0100: burp[25763] nocsr ok
2017-11-03 17:29:30 +0100: burp[25763] SSL is using cipher:
DHE-RSA-AES256-GCM-SHA384 TLSv1.2 Kx=DH       Au=RSA  Enc=AESGCM(256) Mac=AEAD

2017-11-03 17:29:30 +0100: burp[25763] extra_comms_begin
ok:autoupgrade:incexc:orig_client:uname:counters_json:msg:csetproto:
2017-11-03 17:29:30 +0100: burp[25763] Using protocol=1
Backup: 0000001 2017-11-03 17:20:33 +0100 (deletable)
With regex: ^/etc/hosts$
-rw-r--r--  1     0     0     898 2017-11-03 17:05:02 /etc/hosts
2017-11-03 17:29:30 +0100: burp[25763] main socket: Peer closed SSL session
2017-11-03 17:29:30 +0100: burp[25763] List finished ok

** Client: sudo burp -a r -b 1 -r '^/etc/hosts$' -d /tmp
2017-11-03 17:21:08 +0100: burp[25723] Connecting to
burpserver.vagrant.dev:4971
2017-11-03 17:21:08 +0100: burp[25723] auth ok
2017-11-03 17:21:08 +0100: burp[25723] Server version: 2.1.22
2017-11-03 17:21:08 +0100: burp[25723] nocsr ok
2017-11-03 17:21:08 +0100: burp[25723] SSL is using cipher:
DHE-RSA-AES256-GCM-SHA384 TLSv1.2 Kx=DH       Au=RSA  Enc=AESGCM(256) Mac=AEAD

2017-11-03 17:21:08 +0100: burp[25723] extra_comms_begin
ok:autoupgrade:incexc:orig_client:uname:counters_json:msg:csetproto:
2017-11-03 17:21:08 +0100: burp[25723] Using protocol=1
2017-11-03 17:21:08 +0100: burp[25723] doing restore 1:^/etc/hosts$
2017-11-03 17:21:08 +0100: burp[25723] doing restore confirmed

2017-11-03 17:21:08 +0100: burp[25723] zstrm inflate error: -3
2017-11-03 17:21:08 +0100: burp[25723] transfer file returning: -1
2017-11-03 17:21:08 +0100: burp[25723] WARNING: Could not transfer file in:
/tmp/etc/hosts
```

**Comment by grke** [comment]
I can reproduce this problem.

**Comment by grke** [comment]
It turns out that a fix added in 2.1.16 for this somehow got lost in 2.1.18, I guess due to a dodgy merge that I did. :(

**Comment by grke** [comment]
Fixed again in master.
